src/app/content/components/Page/highlightManager.ts

```diff
@@ -81,9 +82,24 @@ const createHighlighter = (services: Omit<HighlightManagerServices, 'highlighter
 
   const highlighter: Highlighter = new Highlighter(services.container, {
     formatMessage: (id: string, style?: string) => intl.formatMessage({ id }, { style }),
-    onClick: (highlight) => onClickHighlight({ ...services, highlighter }, highlight),
-    onFocusIn: (highlight) => services.getProp().focus(highlight.id),
-    onFocusOut: () => services.getProp().clearFocus(),
+    onClick: (highlight) => onFocusHighlight({ ...services, highlighter }, highlight),
+    onFocusIn: (highlight) => onFocusHighlight({ ...services, highlighter }, highlight),
+    onFocusOut: () => {
+      defer(() => {
+        // Do not clear focus from highlight if it was moved to the related card or to another highlight
+        // This function is defered because moving focus is done in a hook inside Card component
+        // which is triggered by a hook in CardWrapper component.
+        // We still want to clear focused highlight if user TAB outside of it, for example to figure link.
+        const activeElement = assertDocument().activeElement;
+        if (
+          activeElement
+          && (activeElement.hasAttribute('data-highlight-card') || activeElement.hasAttribute('data-highlighted'))
```

Comment:
If they click on another card then focus will be moved to this card and it will trigger focusing proper highlight
Clicking "H" will move focus to the highlight related to the focused card